api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/Conv1D.kt

```diff
@@ -116,18 +101,67 @@ public class Conv1D(
 
         val cols = convOutputLength(
             colsCount,
-            kernelSize,
-            paddingInternal,
+            kernelLength,
+            padding,
             strides[1],
             dilations[1]
         )
 
-        return Shape.make(batchSize, cols, filtersInternal.toLong())
+        return Shape.make(batchSize, cols, filters.toLong())
     }
 
     override fun toString(): String =
         "Conv1D(filters=$filters, kernelSize=$kernelSize, strides=${strides.contentToString()}, " +
                 "dilation=${dilations.contentToString()}, activation=$activation, kernelInitializer=$kernelInitializer, " +
                 "biasInitializer=$biasInitializer, kernelShape=${kernel.shape}, biasShape=${bias?.shape}, padding=$padding, " +
                 "biasRegularizer=$biasRegularizer, kernelRegularizer=$kernelRegularizer, activityRegularizer=$activityRegularizer)"
+
+    internal companion object {
```

Comment:
Should these functions be companion functions or could be just private functions in the Conv1D?

Reply:
These functions are used in both `Conv1D` and `Conv1DTransoise` since 1D convolutions are implemented with 2D convolutions.